modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -400,6 +369,11 @@ class PublishDir {
     @CompileStatic
     protected void processFile( Path source, Path destination ) {
 
+        // resolve Fusion symlink if applicable
+        if( FusionHelper.isFusionEnabled(Global.session as Session) )
+            if( source.name in taskInputs )
```

Comment:
Given that we are relaying on Fusion file `.fusion.symlinks` to determine which are the links I think that here is better to use same mechanism. In this way if it's the user that creates a symbolic link it will also be supported.
```suggestion
            if( source.name in getFusionLinks(source.parent) )
```

Reply:
My concern is that this change will greatly increase the number of S3 requests when Fusion is enabled, even if they are small requests.

Currently this code happens only when an output is a symlink via `includeInputs`, which is rare. This change will make Nextflow check the `.fusion.symlinks` file for every task that publishes outputs, whether or not it uses `includeInputs`.

A user-provided symlink with Fusion is probably an anti-pattern anyway, so I would rather not impose the performance cost on the majority of use cases to support it.

On the other hand, it's a small file and probably has little impact on performance, so maybe it doesn't matter.

Reply:
The only case that I imagine that this can be a problem is if they decide to rename an input. But it's a bit a corner case, so make sense to keep as it is to avoid extra S3 requests.